src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/TestManagerBuilder.cs

```diff
@@ -0,0 +1,149 @@
+
+        protected TestManagerBuilder(ITestOutputHelper output)
+        {
+            TestManager = TestFw.TestManager.CreateInstance()
+                .WithXunitTracingInterceptor(output)
+                .WithExtraRmModules(helper => new[]
+                {
+                    helper.RMInsightsModule,
+                })
+                .WithProjectSubfolderForTests("ScenarioTests")
+                .WithCommonPsScripts(new[]
+                {
+                    @"Common.ps1",
+                })
+                .WithNewPsScriptFilename($"{GetType().Name}.ps1")
+                .WithNewMockServerMatcher(
+                    (ignoreResourcesClient, resourceProviders, userAgentsToIgnore) => 
+                        new ResourcesRecordMatcher(ignoreResourcesClient, resourceProviders, userAgentsToIgnore))
+                .WithExtraUserAgentsToIgnore(new Dictionary<string, string>
+                {
+                    {"Microsoft.Azure.Management.ResourceManager.ResourceManagementClient", "2016-07-01"},
+                })
+                .Build();
+
+            var credentials = HttpMockServer.Mode == HttpRecorderMode.Record
+                ? new Func<SubscriptionCloudCredentialsAdapter>(() =>
+                    {
+                        var testEnvironment = TestEnvironmentFactory.GetTestEnvironment();
+                        return new SubscriptionCloudCredentialsAdapter(
+                            testEnvironment.TokenInfo[TokenAudience.Management],
+                            testEnvironment.SubscriptionId);
+                    }) ()
+                : new SubscriptionCloudCredentialsAdapter(
+                    new TokenCredentials("foo"),
+                    Guid.Empty.ToString());
+
+            HttpClientHelperFactory.Instance = new TestHttpClientHelperFactory(credentials);
+        }
+    }
+
+    #region TestHttpClientHelperFactory
+
+    internal class TestHttpClientHelperFactory : HttpClientHelperFactory
+    {
+        /// <summary>
+        /// The subscription cloud credentials.
+        /// </summary>
+        private readonly SubscriptionCloudCredentials credential;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ResourcesController.TestHttpClientHelperFactory"/> class.
+        /// </summary>
+        /// <param name="credentials"></param>
+        public TestHttpClientHelperFactory(SubscriptionCloudCredentials credentials)
+        {
+            credential = credentials;
+        }
+
+        /// <summary>
+        /// Creates new instances of the <see cref="HttpClientHelper"/> class.
+        /// </summary>
+        /// <param name="credentials">The credentials.</param>
+        /// <param name="headerValues">The headers.</param>
+        public override HttpClientHelper CreateHttpClientHelper(SubscriptionCloudCredentials credentials, IEnumerable<ProductInfoHeaderValue> headerValues, Dictionary<string, string> cmdletHeaderValues)
```

Comment:
why not just implement this in the ClientFactory, so there doesn't have to be a special implementation?

Reply:
No modules, other than Resources, need this implementation so far.

Reply:
This could change in the future... As of now only one module uses it, but other modules migh want to sue it (ifte functionality is generic enough).

If the functionality here is super tied to resource manager and we cant foresee anyone ussing this ever, then lets jsut call this class ResourceManagerTestBase.